### ISSUE TITLE
Не удается получить токен.

### ISSUE THREAD

**Issue by 1mavic** [Innim/flutter_login_vk#33]
Не удается получить токен.

1) вызываем метод  vk.logIn(scope: [VKScope.email])
2) открывается приложение vk. Нажимаем там разрешить.
3) возвращаемся в приложение.
4) сразу открывается сафари на странице логина VK.
5) разрешаем там. Страница обновляется. Белый экран.
6) нажимаем готово. Возвращаемся в приложение. Метод логина возвращает null.

Iphone 11. IOs 15.7. Flutter version 3.0.1. flutter_login_vk 2.2.1+1.

**Comment by darkstarx** [comment]
Right, the same behavior, here is my case https://github.com/Innim/flutter_login_vk/issues/36
It appears on the real device and looks like a bug, I just can't sign in.

**Comment by greymag** [comment]
Добрый день. Извиняюсь за поздний ответ. 

Похоже на проблему конфигурации, пожалуйста, сделайте следующее:

1) Проверьте, что приложение настроено как описано в [README](https://github.com/Innim/flutter_login_vk#ios). Обратите внимание на параметр `CFBundleURLTypes`.
2) Если с конфигурацией все в порядке, то проверьте делегат `ios/Runner/AppDelegate.swift`. Убедитесь, что никто не перехватывает обработку открытия в методе `application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool`. Вы можете прислать содержимое файла и я посмотрю все ли там в порядке.
3) Также проверьте любые другие плагины, которые могут перехватывать обработку открытия.

**Comment by 1mavic** [comment]
> 2\. application

спасибо за ответ!

проверил свой AppDelegate.swift
в приложении еще подключен плагин Юкассы:
func application(
    _ application: UIApplication,
    open url: URL,
    sourceApplication: String?, 
    annotation: Any
) -> Bool {
    return YKSdk.shared.handleOpen(
        url: url,
        sourceApplication: sourceApplication
    )
}

Проблема может быть из-за него?

**Comment by greymag** [comment]
Да, именно из-за него =) 

Поменяйте на 

```swift
return YKSdk.shared.handleOpen(
        url: url,
        sourceApplication: sourceApplication
    ) || super.application(app, open: url, sourceApplication: sourceApplication, annotation: annotation)
```

и  аналогично для 
 ``` application(
    _ app: UIApplication,
    open url: URL,
    options: [UIApplication.OpenURLOptionsKey: Any] = [:]
) ``` 
если он у вас объявлен

**Comment by 1mavic** [comment]
> Да, именно из-за него =)
> 
> Поменяйте на
> 
> ```swift
> return YKSdk.shared.handleOpen(
>         url: url,
>         sourceApplication: sourceApplication
>     ) || super.application(app, open: url, sourceApplication: sourceApplication, annotation: annotation)
> ```
> 
> и аналогично для метода
> 
> ```swift
> application(
>     _ app: UIApplication,
>     open url: URL,
>     options: [UIApplication.OpenURLOptionsKey: Any] = [:]
> )
> ```
> 
> если он у вас объявлен

понял, что мне и не нужна функция для Юкассы, убрал - все заработало. Но хорошо бы было решить проблему все-равно.)
В swift совсем не понимаю пока.
Так разве валидно?
```
return YKSdk.shared.handleOpen(
        url: url,
        sourceApplication: sourceApplication
    ) || super.application(app, open: url, sourceApplication: sourceApplication, annotation: annotation)
```
т.е. получаю так:
```

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GeneratedPluginRegistrant.register(with: self)
    FlutterDownloaderPlugin.setPluginRegistrantCallback(registerPlugins)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
  override func application(
        _ application: UIApplication,
        open url: URL,
        sourceApplication: String?,
        annotation: Any
    ) -> Bool {
        return YKSdk.shared.handleOpen(
        url: url,
        sourceApplication: sourceApplication
    ) || super.application(app, open: url, sourceApplication: sourceApplication, annotation: annotation)
    }
    
    @available(iOS 9.0, *)
    override func application(
        _ app: UIApplication,
        open url: URL,
        options: [UIApplication.OpenURLOptionsKey: Any] = [:]
    ) -> Bool {
        return YKSdk.shared.handleOpen(
            url: url,
            sourceApplication: options[UIApplication.OpenURLOptionsKey.sourceApplication] as? String
        )
    }
}
```

**Comment by greymag** [comment]
Да, но во второй функции тоже нужно добавить вызов super, аналогично.
